ui/src/hooks/useFileManage.ts

```diff
@@ -513,71 +523,69 @@ const handleFileUpload = (
-          onFinish: () => void;
-          onError: () => void;
-          onProgress: (e: { percent: number }) => void;
-        }) => {
-          handleFileUpload(<WebSocket>socket, fileList, onProgress, onFinish, onError);
-        }
-      );
-
-      mittBus.on('download-file', ({ path, is_dir }: { path: string; is_dir: boolean }) => {
-        handleFileDownload(<WebSocket>socket, path, is_dir);
-      });
-
-      mittBus.on(
-        'file-manage',
-        ({ path, type, new_name }: { path: string; type: ManageTypes; new_name?: string }) => {
-          switch (type) {
-            case ManageTypes.CREATE: {
-              handleFileCreate(<WebSocket>socket, path);
-              break;
-            }
-            case ManageTypes.CHANGE: {
-              handleChangePath(<WebSocket>socket, path);
-              break;
-            }
-            case ManageTypes.REFRESH: {
-              refresh(<WebSocket>socket, path);
-              break;
-            }
-            case ManageTypes.RENAME: {
-              handleFileRename(<WebSocket>socket, path, new_name!);
-              break;
-            }
-            case ManageTypes.REMOVE: {
-              handleFileRemove(<WebSocket>socket, path);
-              break;
-            }
+    const socket = fileSocketConnection(fileConnectionUrl);
+
+    mittBus.on(
+      'file-upload',
+      ({
+        fileList,
+        onFinish,
+        onError,
+        onProgress
+      }: {
+        fileList: Ref<Array<UploadFileInfo>>;
+        onFinish: () => void;
+        onError: () => void;
+        onProgress: (e: { percent: number }) => void;
+      }) => {
+        handleFileUpload(<WebSocket>socket, fileList, onProgress, onFinish, onError);
+      }
+    );
+
+    mittBus.on('download-file', ({ path, is_dir }: { path: string; is_dir: boolean }) => {
+      handleFileDownload(<WebSocket>socket, path, is_dir);
+    });
+
+    mittBus.on(
+      'file-manage',
+      ({ path, type, new_name }: { path: string; type: ManageTypes; new_name?: string }) => {
+        switch (type) {
+          case ManageTypes.CREATE: {
+            handleFileCreate(<WebSocket>socket, path);
+            break;
+          }
+          case ManageTypes.CHANGE: {
+            handleChangePath(<WebSocket>socket, path);
+            break;
+          }
+          case ManageTypes.REFRESH: {
+            refresh(<WebSocket>socket, path);
+            break;
+          }
+          case ManageTypes.RENAME: {
+            handleFileRename(<WebSocket>socket, path, new_name!);
+            break;
+          }
+          case ManageTypes.REMOVE: {
+            handleFileRemove(<WebSocket>socket, path);
+            break;
           }
         }
-      );
-    }
+      }
+    );
+
+    return socket;
   }
 
-  init();
+  return init();
 };
 
 export const unloadListeners = () => {
   mittBus.off('download-file');
+  mittBus.off('file-upload');
+  mittBus.off('file-manage');
 };
```

Comment:
The code has some minor formatting and style issues that could improve readability:

```typescript
// The above lines of code have been removed due to them potentially being errors or outdated.

@{
    /*
        This section appears incomplete with an "@@" symbol followed by several comments.
    */

@end

```

Additionally, the copyright notice was not included when it used `Copyright (c)`, indicating this may be copyrighted material which needs to include proper attribution and licensing information such as CC-BY-SA. Please note that without proper licensing tags, this software might infringe other's copyrights.